### ISSUE TITLE
Pip install with options failing

### ISSUE THREAD

**Issue by jakehilton** [saltstack/salt#29062]
Trying to do the following:

```
salt-call pip.install boto allow_external=ftputil
```

The output looks like this:
```
[INFO    ] Executing command ['/usr/local/bin/pip2', 'install', 'boto', '--allow-external ftputil'] in directory '/root'
[ERROR   ] Command ['/usr/local/bin/pip2', 'install', 'boto', '--allow-external ftputil'] failed with return code: 2
[ERROR   ] stderr:
Usage:
  pip install [options] <requirement specifier> [package-index-options] ...
  pip install [options] -r <requirements file> [package-index-options] ...
  pip install [options] [-e] <vcs project url> ...
  pip install [options] [-e] <local project path> ...
  pip install [options] <archive url/path> ...

no such option: --allow-external ftputil
[ERROR   ] retcode: 2
local:
    ----------
    pid:
        31997
    retcode:
        2
    stderr:

        Usage:
          pip install [options] <requirement specifier> [package-index-options] ...
          pip install [options] -r <requirements file> [package-index-options] ...
          pip install [options] [-e] <vcs project url> ...
          pip install [options] [-e] <local project path> ...
          pip install [options] <archive url/path> ...

        no such option: --allow-external ftputil
    stdout:
```

If I run the following it succeeds with no errors:
```
# /usr/local/bin/pip2 install boto --allow-external ftputil
Collecting boto
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading boto-2.38.0-py2.py3-none-any.whl (1.3MB)
    100% |████████████████████████████████| 1.3MB 473kB/s
Installing collected packages: boto
Successfully installed boto-2.38.0
```

So the error saying that there is no such option seems to be coming from pip but as we can see it is supported.

Thank you for any and all help.

**Comment by jfindlay** [comment]
@jakehilton, thanks for the report.  This has been fixed by #28757, which will come out in 2015.8.2.

**Comment by jakehilton** [comment]
Awesome. Thank you for the udpate.
